packages/contracts/src/2.0.0/forwarder/interfaces/IForwarderCore.sol

```diff
@@ -25,55 +25,56 @@ import "../../protocol/Exchange/libs/LibFillResults.sol";
 
 contract IForwarderCore {
 
-    /// @dev Market sells ETH for ERC20 tokens, performing fee abstraction if required. This does not support ERC721 tokens. This function is payable
-    ///      and will convert all incoming ETH into WETH and perform the trade on behalf of the caller.
-    ///      This function allows for a deduction of a proportion of incoming ETH sent to the feeRecipient.
-    ///      The caller is sent all tokens from the operation.
-    ///      If the purchased token amount does not meet an acceptable threshold then this function reverts.
-    /// @param orders An array of Order struct containing order specifications.
-    /// @param signatures An array of Proof that order has been created by maker.
-    /// @param feeOrders An array of Order struct containing order specifications for fees.
-    /// @param feeSignatures An array of Proof that order has been created by maker for the fee orders.
-    /// @param feeProportion A proportion deducted off the incoming ETH and sent to feeRecipient. The maximum value for this
-    ///        is 1000, aka 10%. Supports up to 2 decimal places. I.e 0.59% is 59.
-    /// @param feeRecipient An address of the fee recipient whom receives feeProportion of ETH.
-    /// @return FillResults amounts filled and fees paid by maker and taker.
-    function marketSellEthForERC20(
+    /// @dev Purchases as much of orders' makerAssets as possible by selling up to 95% of transaction's ETH value.
+    ///      Any ZRX required to pay fees for primary orders will automatically be purchased by this contract.
+    ///      5% of ETH value is reserved for paying fees to order feeRecipients (in ZRX) and forwarding contract feeRecipient (in ETH).
+    ///      Any ETH not spent will be refunded to sender.
+    /// @param orders Array of order specifications used containing desired makerAsset and WETH as takerAsset. 
+    /// @param signatures Proofs that orders have been created by makers.
+    /// @param feeOrders Array of order specifications containing ZRX as makerAsset and WETH as takerAsset. Used to purchase ZRX for primary order fees.
+    /// @param feeSignatures Proofs that feeOrders have been created by makers.
+    /// @param feePercentage Percentage of WETH sold that will payed as fee to forwarding contract feeRecipient.
+    /// @param feeRecipient Address that will receive ETH when orders are filled.
+    /// @return Amounts filled and fees paid by maker and taker for both sets of orders.
+    function marketSellOrdersWithEth(
```

Comment:
These function names are a bit confusing to me, are they doing the same thing but with a target fill amount in `marketBuyOrdersWithEth`? If so why is one SellOrders and one BuyOrders

Reply:
`marketSell` sells is selling a known amount of `takerAsset` (ETH) and buying an unknown amount of `makerAsset`. This is what you use if you know you want to sell X amount of ETH in advance.
`marketBuy` is selling an unknown amount of `takerAsset` (ETH) and buying a known amount of `makerAsset`. This is what you use if you know you want to buy X amount of `makerAsset` in advance.